### PR TITLE
Beokay refactor master

### PR DESCRIPTION
**Pull request by Alex-Welsh**

### CODE REVIEW

File: a-universe-from-nothing.sh [hunk 1]
```diff
@@ -28,35 +28,19 @@ echo 'Defaults	!fqdn' | sudo tee /etc/sudoers.d/no-fqdn
...
-# Clone Kayobe.
-[[ -d kayobe ]] || git clone https://opendev.org/openstack/kayobe.git -b master
-cd kayobe
+# Clone Beokay.
+[[ -d beokay ]] || git clone https://github.com/stackhpc/beokay.git -b master
 
-# Clone the Tenks repository.
-[[ -d tenks ]] || git clone https://opendev.org/openstack/tenks.git
+# Use Beokay to bootstrap your control host.
+beokay/beokay.py create --base-path ~/deployment --kayobe-repo https://opendev.org/openstack/kayobe.git --kayobe-branch master --kayobe-config-repo https://github.com/stackhpc/a-universe-from-nothing.git --kayobe-config-branch master
```
Comment (markgoddard) [inline a-universe-from-nothing.sh:35]: Is it idempotent?

Comment (Alex-Welsh) [reply a-universe-from-nothing.sh]: It is now!

**Comment by priteau** [comment]
Please squash the commits, as I started to review the oldest ones and left some comments only to realise they were addressed in later commits.

**Comment by Alex-Welsh** [comment]
> Please squash the commits, as I started to review the oldest ones and left some comments only to realise they were addressed in later commits.

Sounds sensible, done

**Comment by Alex-Welsh** [comment]
See also https://github.com/stackhpc/terraform-aufn/pull/19

**Comment by priteau** [comment]
Is the script running successfully to completion? I don't see it sourcing any environment file, so Kolla environment variables would be unset.

File: README.rst [hunk 2]
```diff
@@ -216,12 +203,16 @@ our model development environment, alongside the seed VM.
 
 .. code-block:: console
 
-   # NOTE: Make sure to use ./tenks, since just ‘tenks’ will install via PyPI.
-   export TENKS_CONFIG_PATH=config/src/kayobe-config/tenks.yml
-   ./dev/tenks-deploy-overcloud.sh ./tenks
+   # Set Environment variables for Kayobe dev scripts
+   export KAYOBE_CONFIG_SOURCE_PATH=~/deployment/src/kayobe-config
+   export KAYOBE_VENV_PATH=~/deployment/venvs/kayobe
```
Comment (priteau) [inline README.rst:208]: Is this necessary given you ran `source ~/deployment/env-vars.sh` earlier?

**Comment by Alex-Welsh** [comment]
> Is the script running successfully to completion? I don't see it sourcing any environment file, so Kolla environment variables would be unset.

Good point, no it won't work. I'll fix it soon

File: README.rst (hunk 2 above)
Comment (Alex-Welsh) [reply README.rst:208]: Unfortunately, those are magic vars that are only used in the kayobe dev scripts, so they're not exported by beokay's `env-vars.sh`

Comment (priteau) [reply README.rst:208]: Thanks, I thought they might be part of the `kayobe-env` script, but you're right they aren't.

**Comment by Alex-Welsh** [comment]
I've re-run the whole thing and validated the changes

**Review by MoteHue — APPROVED** [review]
I like this, LGTM

**Pull request merged**